### DISCUSSION TITLE
Getting a list of the entities or a single entity performing operations on them (on it)

### DISCUSSION THREAD

**Discussion by morenostef** [stride3d/stride#2453] [Newcomers]
Hi I need to execute operations on operations on an entity.
with this piece of code below i can read data but i can't modify anything because there is foreach

foreach (Entity ent in Entity.EntityManager)         // lista di tutte le entità
             {
                 Log.Warning(ent.Name);
             }

Is there a way to locate an entity by name directly or with a for loop to change the position or other data of that entity?

**Comment by FloccinaucinihilifilipicatesLongNames** [comment]
I'm a little unclear on the question but if I understand correctly, you need to find an entity by name and then do operations with it like moving it or calling one of the methods of a component, right?

If that's the case, it usually goes like: 
1. Get the scene with the entity/entities: `var MyScene = SceneSystem.SceneInstance.RootScene;`.
 - If the entity in question is in the current scene at runtime, it will be under `SceneSystem.SceneInstance.RootScene` (or one of the child scenes, depending on your setup but I assume it should be like this if you haven't done much with the root scene since I haven't messed around with child scenes yet).
2. Then get the entity from the previous scene: `Entity MyEntity = MyScene.Entities.First(e => e.Name == "Entity name goes here");`
3. Then get (or create) the component you need to modify, for example:
```
TransformComponent MyTRSCompo = MyEntity.Get<TransformComponent>(); // Edit to add: "MyEntity.Transform" does the same thing.
CameraComponent MyCamCompo = MyEntity.GetOrCreate<CameraComponent>();
MyCustomScript MyScriptCompo = MyEntity.Get<MyCustomScript>();
```

Let me know if this helped or if you got more questions.
